lib/twilioconnection.js

```diff
@@ -201,6 +268,9 @@ class TwilioConnection extends StateMachine {
         case 'bad':
           this._handleBad(message);
           break;
+        case 'busy':
+          this._handleBusy(message);
+          break;
         case 'bye':
           // Do nothing.
           break;
```

Comment:
I am surprised that nothing is done in response to a **bye**. The appropriate response would be to close the WebSocket transport. The server should be closing their side but it would be nice to do the same on the client just to be safe.

Reply:
@ceaglest BYE is usually followed by VSG closing the WebSocket with a `1006` error code. I'm not sure if we would want to change the behavior here. What do you do in the C++ SDK?

Reply:
Receiving a bye from the server triggers the client sending a bye and then closing the WebSocket (we only use a 1000 error for now). Since the server's WebSocket close frame usually arrives in the same TCP / TLS packet as the bye the close is handled automatically for us most of the time.

This shouldn't be special case logic though. Is it possible to just have the bye cause a transition to the closed state? This feedback is probably out of scope for the PR at hand.

Reply:
@ceaglest Right now, we depend on the server-side error code to trigger a reconnect. So let's work towards normalizing this behavior in a subsequent PR.